pages/email.tsx

```diff
@@ -60,113 +62,145 @@ export default function Email() {
+                    label={t('given-name.label')}
+                    onChange={formik.handleChange}
+                    value={formik.values.givenName}
+                    errorMessage={
+                      formik.errors.givenName && t(formik.errors.givenName)
+                    }
+                    textRequired={t('common:required')}
+                    required
+                  />
+                  <InputField
+                    id="surname"
+                    name="surname"
+                    label={t('surname.label')}
+                    onChange={formik.handleChange}
+                    value={formik.values.surname}
+                    errorMessage={
+                      formik.errors.surname && t(formik.errors.surname)
+                    }
+                    textRequired={t('common:required')}
+                    required
+                  />
+                  <InputField
+                    id="dateOfBirth"
+                    name="dateOfBirth"
+                    type="date"
+                    label={t('date-of-birth.label')}
+                    onChange={formik.handleChange}
+                    value={formik.values.dateOfBirth}
+                    errorMessage={
+                      formik.errors.dateOfBirth && t(formik.errors.dateOfBirth)
+                    }
+                    textRequired={t('common:required')}
+                    required
+                  />
+                  <div className="flex flex-wrap">
+                    <div className="py-1 pr-2">
+                      <ActionButton
+                        disabled={isEmailEsrfLoading}
+                        type="submit"
+                        text={t('email-esrf')}
+                        style="primary"
+                      />
+                    </div>
+                    <div className="py-1">
+                      <Modal
+                        buttonText={t('common:cancel-modal.cancel-button')}
+                        description={t('common:cancel-modal.description')}
+                        isOpen={modalOpen}
+                        onClick={() => setModalOpen(!modalOpen)}
+                        buttons={[
+                          {
+                            text: t('common:cancel-modal.yes-button'),
+                            onClick: () => Router.push('/landing'),
+                            style: 'primary',
+                            type: 'button',
+                          },
+                          {
+                            text: t('common:cancel-modal.no-button'),
+                            onClick: () => setModalOpen(!modalOpen),
+                            style: 'default',
+                            type: 'button',
+                          },
+                        ]}
+                      />
+                    </div>
+                  </div>
+                </form>
+              </>
+            )
+          }
+        } else {
+          return (
```

Comment:
This return will be call as default rendering because the session storage can only be read client-side. So we might see the screen flicker because it will load with `false` by default then `useEffect` will try to read the value then render accordingly.

Reply:
If ever a search engine crawl the website this page will be indexed with this return component. It makes me rethink how do we want to handle the user consents thing. Maybe using a cookie is the way to go because we can redirect server side if the user didn't consent based on cookie.

Reply:
Tested it and if someone already agreed and refresh the email page then you see the screen flickered from `page-unauth` to `email-form`. I might be a little bit finiky.